service/src/main/java/org/broadinstitute/listener/config/AppConfiguration.java

```diff
@@ -77,14 +79,23 @@ public RelayedHttpRequestProcessor relayedHttpRequestProcessor(
 
   @Bean
   public HybridConnectionListener listener() throws URISyntaxException {
+    Logger logger = LoggerFactory.getLogger(HybridConnectionListener.class);
     RelayConnectionStringBuilder connectionParams =
         new RelayConnectionStringBuilder(properties.getRelayConnectionString());
     TokenProvider tokenProvider =
         TokenProvider.createSharedAccessSignatureTokenProvider(
             connectionParams.getSharedAccessKeyName(), connectionParams.getSharedAccessKey());
-    return new HybridConnectionListener(
-        new URI(connectionParams.getEndpoint().toString() + connectionParams.getEntityPath()),
-        tokenProvider);
+    HybridConnectionListener hybridConnectionListener =
+        new HybridConnectionListener(
+            new URI(connectionParams.getEndpoint().toString() + connectionParams.getEntityPath()),
+            tokenProvider);
+
+    hybridConnectionListener.setOfflineHandler(
+        t -> logger.warn("HybridConnectionListener is now offline"));
+    hybridConnectionListener.setOnlineHandler(
+        () -> logger.warn("HybridConnectionListener is now online"));
```

Comment:
Maybe also have some custom counter metric emitted?

Reply:
I like this idea! It definitely blows up scope of this PR, though, since prometheus and metrics are not currently enabled in this codebase. If we wanted to enable prometheus/metrics, I'd suggest blowing up scope even further and upgrade to Spring Boot 3 first, since a bunch of metrics stuff changed between 2 and 3.